parity/whisper.rs

```diff
@@ -51,7 +51,7 @@ impl PoolHandle for NetPoolHandle {
 	fn relay(&self, message: Message) -> bool {
 		let mut res = false;
 		let mut message = Some(message);
-		self.net.with_proto_context(whisper_net::PROTOCOL_ID, &mut move |ctx| {
+		self.net.with_proto_context(whisper_net::PROTOCOL_ID, &mut |ctx| {
```

Comment:
that's a tricky one. no wonder the tests in the whisper module didn't catch it.

Reply:
I am a little disappointed that rustc doesn't give you a warning for unused_mut when the variable is only copied into a closure.

Minimal playpen: https://play.rust-lang.org/?gist=c66bc6098aeacddb71550878c0d887f9&version=stable

Reply:
https://github.com/rust-lang/rust/issues/47128